hathor/manager.py

```diff
@@ -480,16 +482,19 @@ def _initialize_components(self) -> None:
                     # TODO: deal with invalid tx
                     if tx.can_validate_full():
                         self.tx_storage.add_to_indexes(tx)
-                        assert tx.validate_full(skip_block_weight_verification=skip_block_weight_verification)
+                        assert tx.validate_full(skip_block_weight_verification=skip_block_weight_verification,
+                                                reject_locked_reward=False)
                         self.consensus_algorithm.update(tx)
                         self.tx_storage.indexes.update(tx)
                         if self.tx_storage.indexes.mempool_tips is not None:
                             self.tx_storage.indexes.mempool_tips.update(tx)  # XXX: move to indexes.update
                         if self.tx_storage.indexes.deps is not None:
                             self.sync_v2_step_validations([tx])
+                        self.tx_storage.save_transaction(tx, only_metadata=True)
                     else:
-                        assert tx.validate_basic(skip_block_weight_verification=skip_block_weight_verification)
-                    self.tx_storage.save_transaction(tx, only_metadata=True)
+                        with self.tx_storage.allow_partially_validated_context():
```

Comment:
Why do we need access to partially validated transactions to run a basic validation?

Reply:
Because it might need access to partially validated parents.

Reply:
Since I fixed the cache clearing issue in the test it only needs the context to make the save now, and not the basic validation.

Reply:
Great!